src/sentry/rules/actions/base.py

```diff
@@ -7,20 +7,29 @@
 
 from sentry.eventstore.models import GroupEvent
 from sentry.models.rule import Rule
+from sentry.models.rulefirehistory import RuleFireHistory
 from sentry.rules.base import CallbackFuture, EventState, RuleBase
 
 logger = logging.getLogger("sentry.rules")
 
 
-def instantiate_action(rule: Rule, action):
+def instantiate_action(rule: Rule, action, rule_fire_history: RuleFireHistory | None = None):
+    from sentry.integrations.slack import SlackNotifyServiceAction
     from sentry.rules import rules
 
-    action_cls = rules.get(action["id"])
+    action_id = action["id"]
+    action_cls = rules.get(action_id)
     if action_cls is None:
         logger.warning("Unregistered action %r", action["id"])
         return None
 
-    action_inst = action_cls(rule.project, data=action, rule=rule)
+    if action_id == SlackNotifyServiceAction.id:
+        action_inst = action_cls(
+            rule.project, data=action, rule=rule, rule_fire_history=rule_fire_history
+        )
+    else:
+        action_inst = action_cls(rule.project, data=action, rule=rule)
```

Comment:
I don't think we should have this if statement. If we're going to pass `rule_fire_history` here, it should become part of all the actions and added to the base class. This function should be generic and not know specifics of the actions.

Reply:
Sounds good, I'll update the base class to accept this